### PR TITLE
Discard OTP when high risk card disappears (EXPOSUREAPP-5144)

### PR DESCRIPTION
**Pull request by LukasLechnerDev**
Fulfills the requirement from the Survey story about: 

> When the high-risk card disappears (status change red --> green), a previously generated OTP is discarded.

There are already some unit tests in `RiskLevelChangeDetectorTest`

How to test it manually: 
- Trigger high-risk card (Test Menu => ENF v2 Calculation => INCREASED_RISK_DEFAULT => Calculate Risk Level)
- See high-risk card in HomeMenu; Enter the Risk details screen to see the survey (should currently work on INT)
- Trigger low-risk card (Test Menu => ENF v2 Calculation => LOW_RISK_DEFAULT => Calculate Risk Level)
- Check LogCat => "Discarding one time password for survey about previous high-risk state." should be printed out

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/survey/Surveys.kt [hunk 1]
```diff
@@ -49,7 +49,9 @@ class Surveys @Inject constructor(
     }
 
     suspend fun resetSurvey(type: Type) {
-        // TODO
+        if (type == Type.HIGH_RISK_ENCOUNTER) {
+            settings.clear()
```
Comment (BMItter) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/survey/Surveys.kt:53]: Shouldn't the flow be: OTPRepos clear method calls SurveySettings clear method?

Comment (LukasLechnerDev) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/survey/Surveys.kt]: Yes! Thank you. Changed.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/survey/Surveys.kt [hunk 2]
```diff
@@ -49,7 +50,9 @@ class Surveys @Inject constructor(
     }
 
     suspend fun resetSurvey(type: Type) {
-        // TODO
+        if (type == Type.HIGH_RISK_ENCOUNTER) {
+            oneTimePasswordRepo.clear()
+        }
```
Comment (ralfgehrer) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/survey/Surveys.kt:55]: Minor: Would it be useful to add a Timber to the else case?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/survey/Surveys.kt [hunk 3]
```diff
@@ -49,7 +50,9 @@ class Surveys @Inject constructor(
     }
 
     suspend fun resetSurvey(type: Type) {
```
Comment (ralfgehrer) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/survey/Surveys.kt:52]: It's not a suspend function

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/survey/Surveys.kt [hunk 4]
```diff
@@ -49,7 +50,9 @@ class Surveys @Inject constructor(
...
-        // TODO
+        if (type == Type.HIGH_RISK_ENCOUNTER) {
```
Comment (ralfgehrer) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/survey/Surveys.kt:53]: Minor: please add a Timber to the `if` and `else` case

**Review by ralfgehrer — COMMENTED** [review]
Successfully tested on Pixel (Android 10)

**Review by BMItr — DISMISSED** [review]

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/survey/Surveys.kt (hunk 3 above)
Comment (LukasLechnerDev) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/survey/Surveys.kt]: Thanks, removed!

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/survey/Surveys.kt (hunk 4 above)
Comment (LukasLechnerDev) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/survey/Surveys.kt:54]: Moved the log statement from `RiskLevelChangeDetector` to this if.

**Review by BMItr — APPROVED** [review]

**Review by ralfgehrer — APPROVED** [review]

**Pull request merged**